### PR TITLE
New Guide: Resource Tagging

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* **New Guide:** [Resource Tagging](https://terraform.io/docs/providers/aws/guides/resource-tagging.html)
```

This guide presents a walkthrough of various Terraform AWS Provider functionality relating to resource tags. It especially highlights newer functionality such as provider level ignore tags configuration and individual tag resources. As with other guides originating in this repository, this may make sense to be migrated to the Learn website in the future.

Output from acceptance testing: N/A (documentation only)

### CODE REVIEW

File: website/docs/guides/resource-tagging.html.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+---
+
+# Resource Tagging
+
+Many AWS services implement [resource tags](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html) as an essential part of managing components. These arbitrary key-value pairs can be utilized for billing, ownership, automation, [access control](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_tags.html), and many other use cases. Given that these tags are an important aspect of successfully managing an AWS environment, the Terraform AWS Provider implements additional functionality beyond the typical one-to-one resource lifecycle management for easier and more customized implementations.
+
+-> Not all AWS resources support tagging, which can differ across AWS services and even across resources within the same service. Browse the individual Terraform AWS Provider resource documentation pages for the `tags` argument, to see which support resource tagging. If the AWS API implements tagging support for a resource and its missing from the Terraform AWS Provider resource, a [feature request](https://github.com/terraform-providers/terraform-provider-aws/issues/new?labels=enhancement&template=Feature_Request.md) can be submitted.
```
Comment (ewbankkit) [inline website/docs/guides/resource-tagging.html.md:13]: `its missing` -> `it's missing`

File: website/docs/guides/resource-tagging.html.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    {
+      Name = "MyVPC"
+    },
+  )
+}
+```
+
+## Ignoring Changes to Specific Tags
+
+Systems outside of Terraform may automatically interact with the tagging associated with AWS resources. These external systems may be for administrative purposes, such as a Configuration Management Database, or the tagging may be required functionality for those systems, such as Kubernetes. This section shows methods for prevent Terraform from showing differences for specific tags.
```
Comment (ewbankkit) [inline website/docs/guides/resource-tagging.html.md:66]: `for prevent Terraform` -> `for preventing Terraform`

File: website/docs/guides/resource-tagging.html.md (hunk 1 above)
Comment (bflad) [reply website/docs/guides/resource-tagging.html.md]: Thanks! Fixed.

File: website/docs/guides/resource-tagging.html.md (hunk 2 above)
Comment (bflad) [reply website/docs/guides/resource-tagging.html.md]: Thanks! Fixed.

File: website/docs/guides/resource-tagging.html.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+resource "aws_vpc" "example" {
+  # ... other configuration ...
+
+  tags = {
+    Name = "MyVPC"
+  }
+}
+```
+
+The tags for the resource are wholly managed by Terraform. The only exception to this is tag keys beginning with `aws:` as these are managed by AWS services and usually cannot be edited or deleted. Any non-AWS tags added to the VPC outside of Terraform will be proposed for removal on the next Terraform execution. Missing tags or those with incorrect values from the Terraform configuration will be proposed for addition or update on the next Terraform execution. Advanced patterns that can adjust these behaviors for special use cases, such as Terraform AWS Provider configurations that affects all resources and the ability to manage resource tags for resources not managed by Terraform, can be found later in this guide.
```
[the remark this answers is not in the document]
Comment (bflad) [reply website/docs/guides/resource-tagging.html.md]: Fixed. Thanks!

**Pull request merged**